types/stylelint/index.d.ts

```diff
@@ -194,6 +197,14 @@ declare namespace stylelint {
 	/** @internal */
 	export type CustomSyntax = string | PostCSS.Syntax;
 
+	/** @internal */
+	export type PostProcessFunction = (result: LintResult) => void;
+
+	export type Processor = {
+		name: string;
+		postprocess: PostProcessFunction;
+	};
```

Comment:
[suggest] I have a few suggestions about the type:

- Fix `Processor` to a function type
- Add a warning comment around the experiment
- Inline the internal type

```suggestion
	/**
     * WARNING: This is an experimental feature. The API may change in the future.
     */
	export type Processor = () => {
		name: string;
		postprocess: (result: LintResult) => void;
	};
```